LG-13408 Standardize logging for opted in ipp true/false

jennyverdeyen opened the pull request:
Link to the relevant ticket:
[LG-13408](https://cm-jira.usa.gov/browse/LG-13408)

## 🛠 Summary of changes

Analytics logging for certain events were returning strings "true"/"false" for the property `opted_in_to_in_person_proofing` whereas that property is logged with a boolean value in other events. This change ensures the property is logged as a boolean in these events:

IdV: in person proofing location submitted
IdV: in person proofing location visited
IdV: in person proofing prepare submitted
IdV: in person proofing prepare visited


Provide a checklist of steps to confirm the changes.

- [x] Start the event log watch with `make watch_events` in your terminal
- [x] Trigger each event and observe that `opted_in_to_in_person_proofing` is **true** (Boolean) and not "true" (String), when opted_in_to_in_person_proofing is true
    - [x] Log in through the oidc sinatra app and select "Biometric Comparison"
    - [x] Go through the flow, choosing "Continue in Person" on the how_to_verify page, so `opted_in_to_in_person_proofing` will be **true**
    - [x] Arriving at the Post Office search page, the `"IdV: in person proofing prepare submitted"` and `"IdV: in person proofing location visited"` events will be logged.  Verify that event_properties.opted_in_to_in_person_proofing is **true** (Boolean) for both events.
    - [x] Enter an address in the Post Office search form and click "Search." Select a Post Office location. The `"IdV: in person proofing location submitted"` event will be logged as a result. Verify that event_properties.opted_in_to_in_person_proofing is **true** (Boolean)
    - [x] The `"IdV: in person proofing prepare visited"` is sometimes inconsistent to trigger with opt-in; the page doesn't consistently show up in the flow. One way I have triggered it is by getting to the PO location page, then replacing "#location" with "#prepare" in the url so it reads "http://localhost:3000/verify/document_capture#prepare" - this works for me _from_ the location page, but not necessarily by simply entering that URL from anywhere else. Observe the event is triggered and verify that event_properties.opted_in_to_in_person_proofing is **true** (Boolean)
   
- [x] Trigger each event and observe that `opted_in_to_in_person_proofing` is **false** (Boolean) and not "false" (String), when opted_in_to_in_person_proofing is false
    - [ ] Log in through the oidc sinatra app and select "Biometric Comparison"
    - [x] Go through the flow, choosing "Continue on your phone" on the how_to_verify page. Fail the remote verification, and enter the in person flow where `opted_in_to_in_person_proofing` will now be **false**
    - [x] After failing the remote proofing, you should be taken to "verify/document_capture#prepare" and the `"IdV: in person proofing prepare visited"` will be triggered. Verify that event_properties.opted_in_to_in_person_proofing is **false** (Boolean)
    - [x] Click Continue. Arrive at the Post Office search page and the `"IdV: in person proofing prepare submitted"` and `"IdV: in person proofing location visited"` events will be logged.  Verify that event_properties.opted_in_to_in_person_proofing is **false** (Boolean) for both events.
    - [x] Enter an address in the Post Office search form and click "Search." Select a Post Office location. The `"IdV: in person proofing location submitted"` event will be logged as a result. Verify that event_properties.opted_in_to_in_person_proofing is **false** (Boolean)
    



## Screenshots:

<details>
<summary>Before:</summary>
<img width="376" alt="Screenshot 2024-07-24 at 4 09 11 PM" src="https://github.com/user-attachments/assets/57865f47-0cb5-48b8-85a6-ee2985747615">
<img width="396" alt="Screenshot 2024-07-24 at 4 09 00 PM" src="https://github.com/user-attachments/assets/9882d522-e0fe-40fe-a667-ae9e3738fed6">
<img width="376" alt="Screenshot 2024-07-24 at 4 08 06 PM" src="https://github.com/user-attachments/assets/9d6a8d55-e008-4dcf-8970-e2db2c2db518">
<img width="518" alt="Screenshot 2024-07-24 at 4 08 32 PM" src="https://github.com/user-attachments/assets/4759d6a7-1b03-4e86-8692-fd9b406c13aa">

<img width="368" alt="Screenshot 2024-07-24 at 4 09 46 PM" src="https://github.com/user-attachments/assets/56c6833f-2256-4244-bde5-944f8aa6541d">
<img width="395" alt="Screenshot 2024-07-24 at 4 09 57 PM" src="https://github.com/user-attachments/assets/0fe58fb2-8ae1-48a5-ac59-3d64ba138dad">
<img width="375" alt="Screenshot 2024-07-24 at 4 10 03 PM" src="https://github.com/user-attachments/assets/f67d67b4-49e5-407a-99df-d880006462d2">
<img width="517" alt="Screenshot 2024-07-24 at 4 11 05 PM" src="https://github.com/user-attachments/assets/a7d486b2-cbba-4cb8-bee1-19ab9ceaaca5">



</details>

<details>
<summary>After:</summary>
<img width="522" alt="Screenshot 2024-07-23 at 12 13 59 PM" src="https://github.com/user-attachments/assets/3d96e6b8-a608-4604-985b-e16b5a4f5df8">
<img width="387" alt="Screenshot 2024-07-23 at 12 13 24 PM" src="https://github.com/user-attachments/assets/d9bc40d7-33a0-4643-b8d6-f5e7b42a7bf3">
<img width="396" alt="Screenshot 2024-07-23 at 12 12 58 PM" src="https://github.com/user-attachments/assets/37f59ac2-80fd-42da-bf3b-729fc930c896">
<img width="372" alt="Screenshot 2024-07-24 at 4 06 27 PM" src="https://github.com/user-attachments/assets/ba6dd0e7-36fe-4ff0-936a-0ab383235a62">




<img width="380" alt="Screenshot 2024-07-23 at 12 19 19 PM" src="https://github.com/user-attachments/assets/d145e131-6e76-46d2-b365-25047c09f37f">
<img width="377" alt="Screenshot 2024-07-23 at 12 20 04 PM" src="https://github.com/user-attachments/assets/075888e5-63ed-40d8-a28e-c40f6f8ec415">
<img width="397" alt="Screenshot 2024-07-23 at 12 20 16 PM" src="https://github.com/user-attachments/assets/8c5ede0d-2fe4-4872-af17-8e44e8193c01">
<img width="514" alt="Screenshot 2024-07-23 at 12 20 38 PM" src="https://github.com/user-attachments/assets/83a02b81-a3b9-4545-bd9b-04b9ba272c7e">

</details>

aduth commented on app/services/frontend_logger.rb:
```diff
@@ -37,7 +37,16 @@ def track_event(name, attributes)
-    kwargs(callable).index_with { |key| hash[key.to_s] }
+    kwargs(callable).index_with do |key|
+      value = hash[key.to_s]
+      key == :opted_in_to_in_person_proofing ? convert_to_boolean(value) : value
```
Can we update this from where we're logging in the frontend to pass a boolean instead of a string?

https://github.com/18F/identity-idp/blob/fdf864d1a4c30b5246b0ee96833dd4a2b080fb14/app/javascript/packs/document-capture.tsx#L83

e.g.

```tsx
    opted_in_to_in_person_proofing: optedInToInPersonProofing === 'true',
```

jennyverdeyen replied:
Ah I like this approach. I was looking for where the string was passed in from the front end and didn't find it. Thanks for pointing this out!

jennyverdeyen commented:
@gina-yamada You'll notice I removed the old approach I was discussing with you yesterday in favor of making the conversion from the frontend. Doing it this way also simplified the testing approach since the value is expectedly false instead of nil for the cases in analytics_spec.rb. 

I considered keeping the logger conversion as a future safeguard but ended up removing it in favor of DRYer code... I reasoned that it's unlikely there are other string versions of the value occurring by that point, and the analytics tests could catch that if it were the case. Noting this for posterity, and in case anyone feels the logger conversion is worth keeping.

eileen-nava commented on app/javascript/packs/document-capture.tsx:
```diff
@@ -80,7 +80,7 @@ const trackEvent: typeof baseTrackEvent = (event, payload) => {
     use_alternate_sdk: useAlternateSdk,
     acuant_version: acuantVersion,
-    opted_in_to_in_person_proofing: optedInToInPersonProofing,
+    opted_in_to_in_person_proofing: optedInToInPersonProofing === 'true',
```
👏🏻

eileen-nava reviewed (approved):
I tested and this looks good.  Nice work, Jenny.
